### PR TITLE
fix MAX_INTERMEDIATE_SIZE for DoubleMeanHolder

### PR DESCRIPTION
**Pull request by aP0StAl**
### Description

Fixed the bug with allocating a large amount of memory in the buffer

### CODE REVIEW

**Comment by himanshug** [comment]
+1 after build, thanks , good catch.

How did you get to this? this was really sneaky.

**Comment by aP0StAl** [comment]
@himanshug I found this when I was trying to figure out how to implement a module to solve [this](https://github.com/apache/druid/issues/9539) issue.
I think I found another bug:
https://github.com/apache/druid/blob/55c08e07461e773309c3070ea3891ea304b343de/processing/src/main/java/org/apache/druid/query/aggregation/mean/DoubleMeanAggregatorFactory.java#L148-L171

In some cases, the object may turn out to be byte[] . It's right?

**Comment by aP0StAl** [comment]
@jihoonson Ok. I will add a test tomorrow.

**Comment by himanshug** [comment]
got it, so you were  looking at the code  and knew Long.SIZE wan't number of bytes but bits.

> In some cases, the object may turn out to be byte[] . It's right?

at that point we are never gonna get a byte array.

**Comment by aP0StAl** [comment]
Are you sure? I had an exception before I fixed a similar code (deserialize and finalizeComputation):
`} else if (object instanceof byte[]) {`
`     return DoubleSumVersionFilterHolder.fromBytes((byte[]) object).getSum();`

`Unknown Exception / Unknown object type [[B] / org.apache.druid.java.util.common.IAE`

**Comment by himanshug** [comment]
can you send me the druid query that  reproduces that ?

**Comment by aP0StAl** [comment]
@himanshug It can be either deserialize or finalizeComputation:
`{
  "queryType": "groupBy",
  "dataSource": "test_4",
  "granularity": "all",
  "dimensions": [],
  "aggregations": [
    { "type": "doubleMean", "name": "value", "fieldName": "value"}
  ],
  "intervals": [ "2018-12-01T00:00:00.000/2018-12-02T00:00:00.000" ]
}`

**Comment by himanshug** [comment]
ok, can you please add the byte[] type handling in deserialize(..) and finalizeComputation(..) please. thanks

File: processing/src/main/java/org/apache/druid/query/aggregation/mean/DoubleMeanAggregatorFactory.java [hunk 1]
```diff
@@ -152,6 +152,8 @@ public Object deserialize(Object object)
       return DoubleMeanHolder.fromBytes(StringUtils.decodeBase64(StringUtils.toUtf8((String) object)));
     } else if (object instanceof DoubleMeanHolder) {
       return object;
+    } else if (object instanceof byte[]) {
```
Comment (himanshug) [inline processing/src/main/java/org/apache/druid/query/aggregation/mean/DoubleMeanAggregatorFactory.java]: can we make byte[] be  the firs if check in both  places ?

Comment (aP0StAl) [reply processing/src/main/java/org/apache/druid/query/aggregation/mean/DoubleMeanAggregatorFactory.java]: But DoubleMeanHolder in both places too. Maybe we will replace only String in deserialize(..)?

Comment (himanshug) [reply processing/src/main/java/org/apache/druid/query/aggregation/mean/DoubleMeanAggregatorFactory.java]: I meant, just move the byte[] check as being the first check in if-statement chain as I am expecting that would be the case most of the time.

**Pull request merged**